### PR TITLE
Project GitHub repo link

### PR DESCRIPTION
**Pull request by leonskim**
There's a user feedback: The `GitHub` link at the top of the website always links to the `ir-docs` repository, which is not so intuitive for users who are looking at a project's documentation.

This fixes that.

It looks complicated for such a simple job mainly because of three reasons:
1. Docusaurus doesn't provide `currentPage` sort of reference outside of its own context
2. Automatically generated pages (`generated-index` type pages) are handled differently.  (Added [Comments](https://github.com/infinitered/ir-docs/compare/feat/project-repo-link?expand=1#diff-6914141725ebd631853920c5073b2fc1926061840817bad7a319667532d00345R1-R13) about that)
3. Didn't want to approach with writing a patch

https://github.com/user-attachments/assets/7ebb1b05-13d2-4ac0-994f-3148c7dcacca

### CODE REVIEW

File: scripts/generate-slug.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const baseDocsDir = path.join(__dirname, "../docs");
+
+function updateCategoryFile(filePath, dirName, subDirName) {
+  const data = JSON.parse(fs.readFileSync(filePath, "utf-8"));
+  if (data.link && data.link.type === "generated-index") {
+    const newSlug = `/${dirName}/${subDirName}/list`;
+    data.link.slug = newSlug;
+    fs.writeFileSync(filePath, JSON.stringify(data, null, 2));
+    console.log(`Updated slug for ${filePath}: ${newSlug}`);
```
Comment (mazenchami) [inline scripts/generate-slug.ts:25]: [nits] let's please drop console.logs

File: src/theme/Navbar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const updatedItems = (items[items.length - 1] = {
+    href: repoUrl,
+    label: repoUrl && "GitHub",
+    position: "right",
+  });
+
+  return (
+    <>
+      <Navbar {...props} items={updatedItems} />
+    </>
```
Comment (mazenchami) [inline src/theme/Navbar/index.tsx]: [q] does this need to be wrapped  in `<></>`?

**Review by mazenchami — APPROVED** [review]
impressive work! 🎉 🚀  two questions but shouldn't block you

File: src/theme/Navbar/index.tsx (hunk 2 above)
Comment (leonskim) [reply src/theme/Navbar/index.tsx]: I thought I read it but turns out, no. I must have been hallucinated.

File: scripts/generate-slug.ts (hunk 1 above)
Comment (leonskim) [reply scripts/generate-slug.ts:25]: My intention was to notify the devs that the files have been automatically modified while `yarn install`. But you think that's unnecessary, I'll drop it. :+1:

Comment (mazenchami) [reply scripts/generate-slug.ts:25]: ahh gotcha, since its intentional in that flow, lets keep it! thanks for clarifying

**Comment by trevor-coleman** [comment]
Just saw this -- FYI -- the changes in the docs folder will be overwritten next time those repos are published.

You'll need to open a PR in each repo to make the change to their docs folder.

**Comment by leonskim** [comment]
@trevor-coleman Oh, nice catch! Will do that tomorrow. Thank you sir 😄

**Pull request merged**